jax/_src/export/_export.py

```diff
@@ -711,6 +711,12 @@ def _export_lowered(
   else:
     out_avals_flat = lowered.compile_args["out_avals"]  # type: ignore
 
+  # out_avals come from the Jaxpr, and do not always reflect the out_shardings specification.
```

Comment:
I am confused of where the memory space is stored in a Lowered object. It seems that it could be both in the out_aval and in the out_shardings. For the export we currently use the memory space in the out_aval, and the one in the shardings is ignored (because we export HLO shardings).

I copy here the memory space from the shardings to the aval. But can it happen that the correct memory space is actually in the out aval?

I assume I do not have to do the same for in_avals?